packages/components/src/components/as-dropdown/as-dropdown.tsx

```diff
@@ -78,94 +82,95 @@ export class Dropdown {
-          </button> : '' }
+            <svg viewBox='2 2 12 12' xmlns='http://www.w3.org/2000/svg'><path d='M8,8.58232323 L1.70292632,14.8793969 C1.5421222,15.040201 1.28140721,15.040201 1.12060309,14.8793969 C0.95979897,14.7185928 0.95979897,14.4578778 1.12060309,14.2970737 L7.41767677,8 L1.12060309,1.70292632 C0.95979897,1.5421222 0.95979897,1.28140721 1.12060309,1.12060309 C1.28140721,0.95979897 1.5421222,0.95979897 1.70292632,1.12060309 L8,7.41767677 L14.2970737,1.12060309 C14.4578778,0.95979897 14.7185928,0.95979897 14.8793969,1.12060309 C15.040201,1.28140721 15.040201,1.5421222 14.8793969,1.70292632 L8.58232323,8 L14.8793969,14.2970737 C15.040201,14.4578778 15.040201,14.7185928 14.8793969,14.8793969 C14.7185928,15.040201 14.4578778,15.040201 14.2970737,14.8793969 L8,8.58232323 Z' /></svg>
+          </button> : ''}
       </div>
     );
   }
 
-  /**
-   * Get current selected option
-   *
-   * @returns {string}
-   * @memberof Dropdown
-   */
-  @Method()
-  public async getSelectedOption() {
-    return this.selectedOption;
-  }
-
-  private renderOptions(options: DropdownOption[]) {
+  private _renderOptions(options: DropdownOption[]) {
     return options.map((option) => {
       const buttonClasses = {
         'as-body': true,
-        'is-selected': this.isSelected(option)
+        'is-selected': this._isSelected(option)
       };
 
       return (
         <li class='as-dropdown__list-item' data-value={option.value}>
-          <button class={buttonClasses} onClick={() => this.select(option)}>
+          <button class={buttonClasses} onClick={() => this._select(option)}>
             {option.text || option.value}
           </button>
         </li>
       );
     });
   }
 
-  private isSelected(option) {
+  private _isSelected(option) {
     return option === this.selectedOptionObject;
   }
 
-  private select(option) {
-    if (this.selectedOption === option) {
+  private _select(optionObject) {
+    if (this.selectedOption === optionObject.value) {
       return;
     }
 
-    this.selectedOption = option.value;
-    this.selectedOptionObject = option;
+    this.selectedOption = optionObject.value;
+    this.selectedOptionObject = optionObject;
 
-    this.closeList();
-    this.emitOption();
+    this._closeList();
+    this._emitOption();
   }
 
-  private toggleList() {
+  private _selectFromValue(value) {
+    if (value === undefined || value === null) {
```

Comment:
should we add a tests for the edge case?

Reply:
added